src/components/number/number.jsx

```diff
@@ -23,16 +24,24 @@ function renderAddon(addon, addonClasses, addonSeparator, addonStyle, position)
 
 /**
   Returns the amount of decimals to display when `value >= from_price and value <= to_price`
-  Defaults to `decimals` if no matching tick is found
 */
-function getDecimals(value, decimals, ticks) {
+function getTickDecimals(value, ticks) {
   if (!ticks || !value) {
-    return decimals;
+    return undefined;
   }
 
   const tick = ticks.find(t => value >= t.from_price && value < (t.to_price + t.tick));
 
-  return tick && typeof tick.decimals !== 'undefined' ? tick.decimals : decimals;
+  return tick ? tick.decimals : undefined;
+}
+
+function getFractionDigits(...args) {
```

Comment:
What happens if no argument is finite?

(Eg. what happens when we set maximumFractionDigits={undefined} for the FormattedNumber?)